### ISSUE TITLE
hyprwinwrap: Not Showing Through Transparent Window When Floating

### ISSUE THREAD

**Issue by shadowelite-sec** [hyprwm/hyprland-plugins#197]
HyperWinWrap not showing through transparent window when floating

https://github.com/hyprwm/hyprland-plugins/assets/54718375/c0f05af4-8de3-4408-a045-7eb43f1bca91

**Comment by nnra6864** [comment]
> HyperWinWrap not showing through transparent window when floating
> 2024-07-05_21-48-58.mp4

I think that this is related to the following blur option:
`ignore_opacity = true`
Try disabling it and let me know if it's the same on your end.
I am well aware that the option is required, blur looks disgusting without it, just wanna see if that's the cause.

**Comment by shadowelite-sec** [comment]
> ignore_opacity = true

still the same

**Comment by nnra6864** [comment]
Ah, could be this other thing.
Are you making the terminal transparent in the terminal cfg or via hyprland window rules?
If it's the first, take a look at the issue I created recently:
#167

**Comment by atikiNBTW** [comment]
prob fixed

**Comment by nnra6864** [comment]
> prob fixed

It isn't

**Comment by atikiNBTW** [comment]
> > prob fixed
> 
> It isn't

isn't it the same as #167?

**Comment by nnra6864** [comment]
> isn't it the same as #167?

Nope, #167 refers to hyprwinwrap not being visible through apps that use native transparency.
This issue is hyprwinwrap not being visible through tiled windows.
As I said in #167, I'll make a separate bug report with some additional info soon because I am suspecting this to be a bug on hyprland side and not hyprwinwrap.

**Comment by SchweGELBin** [comment]
Hey, it works when you disable `decoration:blur:new_optimizations`
For anyone still looking for a solution.
Technically a dupe of #91.
Have a great day!

**Comment by nnra6864** [comment]
> Hey, it works when you disable `decoration:blur:new_optimizations` For anyone still looking for a solution. Technically a dupe of #91. Have a great day!

Yup, already mentioned the solution in #212

**Comment by SchweGELBin** [comment]
@nnra6864 Thank you very much! I didn't find your issue here. Can you tell me which program/config you use for hyprwinwrap? It looks pretty nice.

**Comment by nnra6864** [comment]
> @nnra6864 Thank you very much! I didn't find your issue here. Can you tell me which program/config you use for hyprwinwrap? It looks pretty nice.

Sure, I used to use [swww](https://github.com/LGFae/swww) + [GLava](https://github.com/jarcode-foss/glava) with my own [config](https://github.com/nnra6864/Ricerland/tree/Arch/glava).
However, I decided to make my own visualizer app, [Nisualizer](https://github.com/nnra6864/Nisualizer), and I'm on that now.
It's still very early in development and I'd highly advise against using it but you are free to take a look.

**Comment by SchweGELBin** [comment]
Thank you for your response. Your app looks very promising.
I'm looking for the circular version instead of the bars version of cava.
GLava is pretty dated and I just can't figure out, how to do it.
Can you help me with it?

Sorry that I am asking here; it's probably not the right place.

Have a great day!

**Comment by nnra6864** [comment]
> Thank you for your response. Your app looks very promising. I'm looking for the circular version instead of the bars version of cava. GLava is pretty dated and I just can't figure out, how to do it. Can you help me with it?
> 
> Sorry that I am asking here; it's probably not the right place.
> 
> Have a great day!

Sure, I'd be up to help you set it up.
You can add me on discord `_nnra` so we don't spam this issue.

**Comment by fdev31** [comment]
Here it randomly works sometimes, I can't figure the sequence to "enable" it, but most of the time I get nothing in the background instead of the hyprwinwrap content.

**Comment by SchweGELBin** [comment]
Hey, I don't really have any problems with hyprwinwrap anymore. Here are some parts of my config:
I didn't test it, but something like this should work.

``` nix
{ pkgs, ... }:
{
    wayland.windowManager.hyprland = {
        plugins = with pkgs.hyprlandPlugins; [ hyprwinwrap ];
        settings = {
            bindd = [ "SUPER, B, Toggle Cava Background, exec, avabg c t" ];
            decoration.blur.new_optimizations = false;
            exec-once = [ "avabg c" ];
            plugin.hyprwinwrap.class = "hyprbg";
        };
    };
    
    xdg.configFile."alacritty/cava.toml".text = ''
        [window]
        opacity = 0.0
    '';
    
    home.packages = [
        (pkgs.writeShellScriptBin "avabg" ''
            pkill cava
            pkill glava
            if [[ $2 == "t" && ! -z $(pgrep avabg-instance) ]]; then
              pkill avabg-instance
            else
              avabg-instance $1
            fi
          '')
    
          (pkgs.writeShellScriptBin "avabg-instance" ''
            case $1 in
            c) # Cava
              alacritty --class 'hyprbg' --config-file '${vars.user.home}/.config/alacritty/cava.toml' -e 'cava'
            ;;
            gl) # GLava
              echo "I don't know how to set the class for GLava"
            ;;
            esac
          '')
    ];
}
```

**Comment by nnra6864** [comment]
> Here it randomly works sometimes, I can't figure the sequence to "enable" it, but most of the time I get nothing in the background instead of the hyprwinwrap content. It also creates random issues (focus, windows visibility, ...) after some time (also quite random).

Did you try the solution @SchweGELBin provided above?
Add this to your config:
`decoration:blur:new_optimizations = false`
